crates/cheatcodes/src/defs/mod.rs

```diff
@@ -0,0 +1,181 @@
+//! Cheatcode definitions.
+
+mod hevm;
+pub use hevm::HEVM;
+
+use alloy_sol_types::SolCall;
+use serde::{Deserialize, Serialize};
+
+/// Stable specification of a single cheatcode.
+#[derive(Serialize, Deserialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct Cheatcode<'a> {
+    // Automatically-generated fields.
+    /// The cheatcode's identifier. Corresponds to the function's name,
+    /// thus is not unique if the function is overloaded.
+    pub id: &'a str,
+    /// The full Solidity function signature, including parameter names, visibility, etc.
+    /// This is just for reference, and is identical to the function declaration.
+    pub full_signature: &'a str,
+    // TODO: add visibility, state mutability...?
```

Comment:
All will have visibility of `external`, but for mutability here's the way we define when to use pure vs. view vs. neither in forge-std: https://github.com/foundry-rs/forge-std/blob/f73c73d2018eb6a111f35e4dae7b4f27401e9421/src/Vm.sol#L6-L10. Might be useful to include a visibility field in the schema for consistency—there's been some recent PRs correcting visibility of cheats in forge-std that probably didn't make it upstream to foundry

Reply:
Added here <https://github.com/foundry-rs/foundry/pull/5998/commits/d68edfbcfce4855602726760255edbeab3e9fa24>